components/date_picker/DatePicker.js

```diff
@@ -134,8 +135,8 @@ const factory = (Input, DatePickerDialog) => {
             maxDate={maxDate}
             minDate={minDate}
             name={this.props.name}
+            onDismiss={onDismiss || this.handleDismiss}
```

Comment:
I would rather use the `handleDismiss` always and then call `onDismiss` from the same handler in case it exists. Looks more clear if the handler is always called.